spec/solver_spec.rb

```diff
@@ -0,0 +1,33 @@
+require_relative '../solver'
+
+describe Solver do
+  context 'Testing solver methods' do
+    before :each do
+      @solver = Solver.new
+    end
+
+    it 'Factorial returns correct answer' do
+      expect(@solver.factorial(4)).to eq 24
+    end
```

Comment:
- [ ] Please add all test cases for Getting the factorial of a number to cover all possibilities so add a test case for the factorial of a negative number and another one for the factorial of number 0 (zero). and then add those cases to your factorial class